utils/municipality.ts

```diff
@@ -0,0 +1,365 @@
+            )
+          )
+      )
+    );
+
+const calculateMunicipalityPath = (codiceCatastale: string) =>
+  `${MUNICIPALITIES_CONTAINER_NAME}/${codiceCatastale.charAt(
+    0
+  )}/${codiceCatastale.charAt(1)}` as NonEmptyString;
+
+export const serializeMunicipalityToJson = (
+  blobService: BlobService,
+  serializableMunicipality: ISerializableMunicipality
+): TaskEither<Error, Option<BlobService.BlobResult>> =>
+  CodiceCatastale.decode(serializableMunicipality.codiceCatastale).fold(
+    errs =>
+      fromLeft(
+        new Error(`Cannot decode CodiceCatastale| ${readableReport(errs)}`)
+      ),
+    codiceCatastaleValue =>
+      writeBlobFromJson(
+        blobService,
+        calculateMunicipalityPath(codiceCatastaleValue),
+        `${codiceCatastaleValue}.json` as NonEmptyString,
+        JSON.stringify(serializableMunicipality.municipality) as NonEmptyString
+      )
+  );
+
+export const exportAbolishedMunicipality = (
+  blobService: BlobService
+): TaskEither<Error, ReadonlyArray<Option<BlobService.BlobResult>>> =>
+  loadMunicipalityToCatastale(blobService)
+    .chain(municipalityToCatastale =>
+      loadAbolishedMunicipalities(blobService, municipalityToCatastale)
+    )
+    .chain(abolishedMunicipalities =>
+      array.sequence(taskEither)(
+        abolishedMunicipalities.map(municipality =>
+          serializeMunicipalityToJson(blobService, municipality)
+        )
+      )
+    );
+
+export const exportCurrentMunicipalities = (
+  blobService: BlobService
+): TaskEither<Error, ReadonlyArray<Option<BlobService.BlobResult>>> =>
+  getCsvFromURL(ITALIAN_MUNICIPALITIES_URL)
+    .chain(csvContent =>
+      parseCsv(csvContent, currentMunicipalitiesParserOption)
+    )
+    .chain(result =>
+      array.sequence(taskEither)(
+        result.map(r =>
+          decodeMunicipality(r).fold(
+            e =>
+              fromLeft(
+                new Error(`Cannot decode current municipality| ${e.toString()}`)
+              ),
+            municipality =>
+              serializeMunicipalityToJson(blobService, {
+                codiceCatastale: r[
+                  CODICE_CATASTALE_INDEXES.currentMunicipality
+                ] as NonEmptyString,
+                municipality
+              })
+          )
+        )
+      )
+    );
+
+export const exportForeignMunicipalities = (
+  blobService: BlobService
+): TaskEither<Error, ReadonlyArray<Option<BlobService.BlobResult>>> =>
+  getFileFromBlob(
+    blobService,
+    MUNICIPALITIES_CONTAINER_NAME,
+    FOREIGN_COUNTRIES_BLOB_NAME
+  )
+    .chain(maybeCsv =>
+      maybeCsv.foldL(
+        () =>
+          fromLeft<Error, StringMatrix>(
+            new Error("Elenco codici e denominazioni is empty")
+          ),
+        csv => parseCsv(csv, optionCsvParseForeignCountries)
+      )
+    )
+    .chain(result =>
+      array.sequence(taskEither)(
+        filterEmptyCodiceCatastaleRecords(result, 9 as NonNegativeInteger).map(
```

Comment:
why "9" ? I guess the index value must be hardcoded into `filterEmptyCodiceCatastaleRecords` instead of being taken as paramter

Reply:
agree, just refactor